### PR TITLE
Cyra

### PR DESCRIPTION
**Pull request by cyraalesha**
Added lab 2. Not sure if the flake is entirely correct but it seems to build successfully and created a flake.lock

### CODE REVIEW

File: Cyra/lab2/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.14)
+project(nixintro)
+include_directories(include)
```
Comment (RCMast3r) [inline Cyra/lab2/CMakeLists.txt:3]: do not use include_directories, only use target_include_directories and include directories per each target, not globally within a cmakelists

File: Cyra/lab2/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+
+add_library(hellolib SHARED src/hellolib.cpp)
```
Comment (RCMast3r) [inline Cyra/lab2/CMakeLists.txt:6]: this doesnt appear to be the correctly named source file, cmake is case-sensitive for these file names

File: Cyra/lab2/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+add_executable(nixtest src/main.cpp)
+
```
Comment (RCMast3r) [inline Cyra/lab2/CMakeLists.txt:8]: where is your install phase for lab2? this is required by nix

File: Cyra/lab2/flake.nix [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      aarch64-darwin = let
+        pkgs = nixpkgs.legacyPackages.aarch64-darwin;
+      in
+      {
+        hello = pkgs.hello;
+        default = pkgs.callPackage ./default.nix {};
+      };
+    };
+  };
+}
```
Comment (RCMast3r) [inline Cyra/lab2/flake.nix:19]: this flake is missing a proper use of nix packages, see:

https://github.com/nixvital/nix-based-cpp-starterkit/blob/main/flake.nix

**Review by RCMast3r — CHANGES_REQUESTED** [review]
it doesnt loo

**Comment by cyraalesha** [comment]
Updated

**Review by RCMast3r — APPROVED** [review]

**Pull request merged**